server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/reservedstate/ReservedRepositoryAction.java

```diff
@@ -36,7 +35,7 @@
  * It is used by the ReservedClusterStateService to add/update or remove snapshot repositories. Typical usage
  * for this action is in the context of file based settings.
  */
-public class ReservedRepositoryAction implements ReservedClusterStateHandler<List<PutRepositoryRequest>> {
+public class ReservedRepositoryAction implements ReservedProjectStateHandler<List<PutRepositoryRequest>> {
```

Comment:
Basic question. In a non-MP stateless cluster, this is still sufficient because now the cluster back up repo falls under the "default" project?

Reply:
Yes. Additionally, we have [ProjectClusterStateHandlerAdapter](https://github.com/elastic/elasticsearch/blob/3b1397734337d11c74128d206d8eaa431185178c/server/src/main/java/org/elasticsearch/reservedstate/service/ProjectClusterStateHandlerAdapter.java#L22) which reads the reserved state from the main `settings.json` file in a non-MP setup.

Reply:
This is correct, no need to explicitly handle the non-MP case.